containers/podman-desktop
# feat(website): add blog social media banner

### What does this PR do?

This PR sets explicit featured images for individual blog posts (the most recent 5, to be exact) so when those blog post links are shared on social media, etc., they will display the image for that post in the preview.

Right now, no image is defined, so they show up blank and sad.

I also attempt to set a default, general "Podman Desktop" graphic banner to show up in the case that we don't define a featured image for any individual blog post. Also, this general banner should show up if any link on podman-desktop.io is shared and generates a preview card.

(I say "attempt" because for whatever reason, I couldn't test the default locally without using https://localhost:3000/img/banner_podman-desktop.png as the image url value... of course I'm not going to hard code that... I committed the relative path 'img/banner_podman-desktop.png' [as demonstrated in the upstream docs](https://docusaurus.io/docs/api/themes/configuration#meta-image) and hoping it words in 'prod')

### Screenshot/screencast of this PR

#### BEFORE 😭

![Screenshot from 2023-08-21 11-22-50](https://github.com/containers/podman-desktop/assets/799683/b8faabe0-11fb-4caa-98f8-d49cebf97e01)

#### AFTER 🥰

![Screenshot from 2023-08-21 14-06-43](https://github.com/containers/podman-desktop/assets/799683/08f83f76-2a94-4320-9574-3e0ee79ceeaa)


### What issues does this PR fix or reference?

https://github.com/containers/podman-desktop/issues/3613

### How to test this PR?

- `yarn` && `yarn start` in the website directory of the repo
- you need a social media previewer thing that will work with local URLs. i used this one in chrome: https://chrome.google.com/webstore/detail/social-share-preview/ggnikicjfklimmffbkhknndafpdlabib
- Pop up the social media previewer tool and visit `http://localhost:3000`
- Anywhere you go on the website should have this in the previewer:
![Screenshot from 2023-08-29 09-11-24](https://github.com/containers/podman-desktop/assets/799683/880e6bcd-7393-42ae-9f2b-c4a864c7e628)
- Then visit the blog... the 5 most recent posts should have banners in the social media previewer that match the image in the blog post eg
![Screenshot from 2023-08-29 09-11-34](https://github.com/containers/podman-desktop/assets/799683/2a5742f3-9aec-4fea-be8e-a766109dfb4e)
